jupyter/README.md

```diff
@@ -0,0 +1,138 @@
+
+   - Run the scripts as described [here](https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html). Most likely, only the following command will be needed (assuming the script is in your CWD)- `bash desired_script_name`.
+
+2. **Create conda environment**
+
+  - One of the many ways is to use a **yaml** file that contains a comprehensive list of packages that you need. For example, once can use the file [py38.yaml](../conda/py38.yaml) in this same repository.
+
+  - Use this command to create the conda environment -
+  ```
+  conda env create -f py38.yaml
+  ```
+
+3. **Set a password for jupyter notebook**
+
+  - When you run a jupyter server for the first time you will have to set up a password, and deal with some other specifics. While the instructions are summarized here, you can find more information [here](https://jupyter-notebook.readthedocs.io/en/stable/public_server.html).
+
+  - Notebook configuration file
+
+    The following file should be present `.jupyter/jupyter_notebook_config.py` in your installation directory (most likely your home directory). If not use this command to generate one -
+    ```
+    jupyter notebook --generate-config
+    ```
+
+  - Set your password
+
+    There are two ways to do this.
+
+    - Automatic
+      ```
+      jupyter notebook password
+      Enter password:  ****
+      Verify password: ****
+      ```
+
+      This will modify the json file in `.jupyter`.
+
+    - Manual
+
+      Alternatively, you can convert your password into a hash key and add it manually to `.jupyter/jupyter_notebook_config.py`.
+
+      ```
+      $ ipython
+      In [1]: from notebook.auth import passwd
+      In [2]: passwd()
+      Enter password: ****
+      Verify password: ****
+      Out[2]: 'your_hashed_password'
+      ```
+
+      Add hashed password to your notebook configuration file.
+
+      Find the `c.NotebookApp.password` field in `.jupyter/jupyter_notebook_config.py`, which might be commented out. Modify the line as follows -
+      ```
+      c.NotebookApp.password = u'your_hashed_password'
+      ```
+
+    Automatic password setup will store the hash in `jupyter_notebook_config.json` while this method stores the hash in `jupyter_notebook_config.py`. The `.json` configuration options take precedence over the `.py` one, thus the manual password may not take effect if the `.json` file has a password set.
+
+  - Use SSL for encrypted communication
+
+    Generate a self signed certificate and key.
+
+    ```
+    openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout mykey.key -out mycert.pem
+    ```
+
+    You can start the notebook to communicate via a secure protocol mode by setting the certfile option to your self-signed certificate, i.e. mycert.pem, with the command:
+    ```
+    jupyter notebook --certfile=mycert.pem --keyfile mykey.key
```

Comment:
Maybe worth mentioning that this is not necessary if `mycert.pem` and `mykey.key` are added to the `jupyter_notebook_config.py` file ... is that right? I actually never did any of the encryption steps 🤫

Reply:
Yep, I think this is correct.